### PR TITLE
[esp32] reduce default event queue size

### PR DESCRIPTION
**Pull request by gjc13**
#### Problem

The event queue is way too large by default, occupying 6.4 KB.

#### Change overview

Reduce the default event queue size to 25. This value has teen long used in qpg and efr platforms.

#### Testing

Build all-cluster-app on ESP32 and pair with chip-device-ctrl. The pairing flow can still work.

### CODE REVIEW

File: config/esp32/components/chip/Kconfig [hunk 1]
```diff
@@ -350,7 +350,7 @@ menu "CHIP Device Layer"
         config MAX_EVENT_QUEUE_SIZE
             int "Max Event Queue Size"
             range 0 65535
-            default 100
+            default 20
```
Comment (dhrishi) [inline config/esp32/components/chip/Kconfig:353]: @gjc13  Is this the event queue for the CHIP task? Also, is 20 the optimized value?

Comment (gjc13) [reply config/esp32/components/chip/Kconfig:353]: Yes, this is the event queue in the device layer. I'll try 5-10 as well.

File: src/include/platform/CHIPDeviceConfig.h [hunk 2]
```diff
@@ -88,7 +88,7 @@
  * The maximum number of events that can be held in the chip Platform event queue.
  */
 #ifndef CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE
-#define CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE 100
+#define CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE 25
```
Comment (dhrishi) [inline src/include/platform/CHIPDeviceConfig.h:91]: I am not really sure if we should change this value generically (for all the platforms). Else, LGTM

**Review by dhrishi — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/include/platform/CHIPDeviceConfig.h (hunk 2 above)
Comment (tcarmelveilleux) [reply src/include/platform/CHIPDeviceConfig.h:91]: Agree that this should not be changed by *platform*. This setting actually should be *by product*.

**Comment by andy31415** [comment]
Why is the event queue too large? 

It is too large for QPG  where it was reduced to save RAM, however if the larger devices support it, is there any reason to reduce it? Please add more details about why this change is required in the summary.

**Comment by gjc13** [comment]
@andy31415 @tcarmelveilleux The defautl value in the platform config is left default and the value in ESP32 Kconfig is modified since we do have received complaints on heap usage.

**Review by LuDuda — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**